include/dem/uniform_insertion.h

```diff
@@ -40,17 +40,13 @@ template <int dim>
 class UniformInsertion : public Insertion<dim>
 {
 public:
-  /**
-   * The constructor to the insertion class calculates the maximum number of
-   * particles (maximum_particle_number) which can be inserted in the insertion
-   * box and prints a warning if the number of particles to be inserted at this
-   * step exceeds this calculated number Note that in the current version only
-   * the insertion in a rectangular box is defined
-   *
-   * @param dem_parameters DEM parameters declared in the .prm file
-   */
-  UniformInsertion<dim>(const DEMSolverParameters<dim> &dem_parameters);
+  UniformInsertion<dim>(const DEMSolverParameters<dim> &dem_parameters,
+                        unsigned int &                  inserted_this_step,
+                        unsigned int &                  nx,
+                        unsigned int &                  ny,
+                        unsigned int &                  nz);
```

Comment:
I think it would be a lot better if the constructor took instead of the nx ny and nz and inserted_this_step only the dem parameters it needs (so only the Parameters::insertion). Then you can construct whatever you need from this. I don't understand why the constructor requires the inserted_this_step.